PULL_REQUEST_TEMPLATE.md

```diff
@@ -0,0 +1,31 @@
+Please try to adhere to the guidelines below as much as possible when submitting your pull request.
+
+- Please verify that your code satisfies the
+[code/documentation quality guidelines](http://scikit-learn.org/stable/developers/contributing.html#coding-guidelines).
+- Please prefix the title of your pull request with `[MRG]` if the
+contribution is complete and should be subjected to a detailed review.
+
+Otherwise:
+- Incomplete contributions should be prefixed `[WIP]` to indicate a work
+in progress (and changed to `[MRG]` when it matures). WIPs may be useful
+to: indicate you are working on something to avoid duplicated work,
+request broad review of functionality or API, or seek collaborators.
+WIPs often benefit from the inclusion of a
+[task list](https://github.com/blog/1375-task-lists-in-gfm-issues-pulls-comments)
+in the PR description.
+- Documentation and high-coverage tests are necessary for enhancements to
+be accepted.
+- If you are adding an enhancement, you may wish to provide evidence for
+its benefit with distinguishing examples in the code and benchmarks
+in the PR discussion.
```

Comment:
I would definitely add somewhere that if you are addressing an issue, the title of the PR should be describing the issue, but the text of the PR should have #issuenumber in it, so that a link is created. We often get titles like "fixing #1234" which I don't find very helpful for browsing the pull requests, and also they don't create a link to the issue.